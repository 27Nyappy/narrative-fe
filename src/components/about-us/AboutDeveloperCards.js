import React from 'react';
import PropTypes from 'prop-types';
import AboutDeveloperCard from './AboutDeveloperCard'; 

const AboutDeveloperCards = ({ aboutDeveloperCards }) => {
  const aboutDeveloperElements = aboutDeveloperCards.map(card => (
    <div key={card.developerName}>
      <AboutDeveloperCard developerName={card.developerName} developerImage={card.developerImage} developerBio={card.developerBio}/>
    </div>
  ));

  return (
    <section>
      {aboutDeveloperElements}
    </section>
  );
};

AboutDeveloperCards.propTypes = {
  aboutDeveloperCards: PropTypes.arrayOf(PropTypes.shape({
    developerName: PropTypes.string.isRequired,
    developerImage: PropTypes.string.isRequired,
    developerBio: PropTypes.string.isRequired
  })).isRequired
};

export default AboutDeveloperCards;