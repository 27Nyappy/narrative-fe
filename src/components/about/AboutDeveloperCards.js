import React from 'react';
import PropTypes from 'prop-types';
import AboutDeveloperCard from './AboutDeveloperCard'; 

const AboutDeveloperCards = ({ aboutDeveloperCards }) => {
  const aboutDeveloperElements = aboutDeveloperCards.map(card => (
    <div key={card.name}>
      <AboutDeveloperCard developerName={card.name} developerImage={card.image} developerBio={card.bio}/>
    </div>
  ));

  return (
    <section>
      {aboutDeveloperElements}
    </section>
  );
};

AboutDeveloperCards.propTypes = {
  aboutDeveloperCards: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired
  }))
};

export default AboutDeveloperCards;