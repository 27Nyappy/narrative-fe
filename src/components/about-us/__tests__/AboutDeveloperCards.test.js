import React from 'react';
import { shallow } from 'enzyme';
import AboutDeveloperCards from '../AboutDeveloperCard';

describe('AboutDeveloperCards component', () => {
  it('renders a list of about developer cards', () => {
    const developers = [
      {
        developerName: 'developer 1',
        developerImage: 'developer1.jpg',
        developerBio: 'bio 1'
      },
      {
        developerName: 'developer 2',
        developerImage: 'developer2.jpg',
        developerBio: 'bio 2'
      },
      {
        developerName: 'developer 3',
        developerImage: 'developer3.jpg',
        developerBio: 'bio 3'
      }
    ];

    const wrapper = shallow(<AboutDeveloperCards AboutDeveloperCards={developers}/>);
    expect(wrapper).toMatchSnapshot();
  });
});