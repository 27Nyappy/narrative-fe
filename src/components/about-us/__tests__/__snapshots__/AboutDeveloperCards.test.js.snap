// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutDeveloperCards component renders a list of about developer cards 1`] = `
<Fragment>
  <h2 />
  <div>
    <img />
    <p />
  </div>
</Fragment>
`;
